### PR TITLE
Feature/process include option

### PR DESCRIPTION
**Pull request by kovalenko0**

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -1,30 +1,42 @@
...
 
       ${tsconfigPath}
 
     Consider either passing an explicit list of files or adding the "files" key.
   `;
 
+  if (!files && include) {
+    files = include
+      .filter(isGlob)
+      .map((glob) => globFs({}).readdirSync(glob))
```
Comment (pzavolinsky) [inline src/app.ts]: Can we make this `.map(globFs({}).readdirSync)`?

File: src/app.ts [hunk 2]
```diff
@@ -1,30 +1,42 @@
...
     The tsconfig does not contain a "files" key:
 
       ${tsconfigPath}
 
     Consider either passing an explicit list of files or adding the "files" key.
   `;
 
+  if (!files && include) {
+    files = include
+      .filter(isGlob)
```
Comment (pzavolinsky) [inline src/app.ts]: According to the [docs](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html) the `include` section can only be a list of globs, so maybe we could just map them right away without the filter + concat + dedup below.

What do you think? Is there an example where something in the `include` might break `glob-fs`?

File: src/app.ts [hunk 3]
```diff
@@ -35,8 +47,28 @@ export default (tsconfigPath:string, files?:string[]) => {
   return analyze(
     parseFiles(
       dirname(tsconfigPath),
-      files || tsConfig.files,
+      files || tsConfig.files || [],
```
Comment (pzavolinsky) [inline src/app.ts]: `tsConfig.files` cannot be `undefined` so no need to `|| []`.

**Comment by mparpaillon** [comment]
Any news on this feature ? Is it possible to use "include" instead of "files" ?

**Comment by kovalenko0** [comment]
@mparpaillon as far as I remember, the code in pull request worked. So, you might use that. 
If you're using VS Code, there is a typescript.referencesCodeLens.enabled setting, that shows how many references there are to a particular symbol in the file.

**Comment by pzavolinsky** [comment]
@mparpaillon not ideal, but a workaround that might work for you is to pass the files to be checked as an argument to `ts-unused-exports`, as in:

```js
ts-unused-exports path/to/tsconfig.json file1.ts ...
```

**Comment by mparpaillon** [comment]
Not ideal indeed... that's too bad. But thanks

**Comment by pzavolinsky** [comment]
@mparpaillon I've updated the code to avoid any custom parsing of `tsconfig.json` and instead delegate all that work to the TS compiler. As a result, `include`, `exclude`, etc. should be working. Check version `2.0.8`.

Also, thanks @kovalenko0 for this PR, I'm sorry it took me so long to tackle this issue.

Cheers!

**Pull request closed without merging**